### PR TITLE
Ensure combined epic subscription is disposed with store

### LINKED ISSUE #12: Should dispose of the epic subscription
Just discovered that the combined epic stream os never closed when the store is disposed.

https://github.com/johanflint/rxstore.dart/blob/6a28c5e7ac1f309d6c02509d22f1c63b2d098bfd/packages/rxstore/lib/src/store.dart#L72

To reproduce this, create an epic with a periodic stream. Even when the store is disposed, the stream never ends.

```dart
Epic<AppState> periodicEpic() => (actions, state) =>
    actions.whereType<MyAction>().switchMap((action) {
      return Stream<Action>.periodic(
        const Duration(seconds: 1),
        (_) => const NoopAction(),
      ).doOnCancel(() => print('canceled periodic stream'));
    }).doOnCancel(() => print('canceled periodic epic'));
```

I propose to keep a reference to the combined epic and dispose when the store is disposed

### PR DESCRIPTION
**Pull request by jogboms**
Closes #12

### CODE REVIEW

File: packages/rxstore/lib/src/store.dart [hunk 1]
```diff
@@ -78,6 +78,8 @@ class Store<State> {
   final BehaviorSubject<State> _changeSubject;
   final PublishSubject<Action> _dispatchSubject;
 
+  StreamSubscription? _epicSubscription;
```
Comment (jogboms) [inline packages/rxstore/lib/src/store.dart:81]:
```suggestion
late final StreamSubscription? _epicSubscription;
```

Comment (johanflint) [inline packages/rxstore/lib/src/store.dart:81]:
```suggestion
   late final StreamSubscription? _epicSubscription;
```

**Review by johanflint — APPROVED** [review]
LGTM

File: packages/rxstore/lib/src/store.dart (hunk 1 above)
Comment (johanflint) [reply packages/rxstore/lib/src/store.dart:81]: Doesn't make it better :).

**Pull request merged**